xbmc/filesystem/CurlFile.cpp

```diff
@@ -334,10 +334,16 @@ long CCurlFile::CReadState::Connect(unsigned int size)
   // read some data in to try and obtain the length
   // maybe there's a better way to get this info??
   m_stillRunning = 1;
+
+  // (Try to) fill buffer
   if (!FillBuffer(1))
   {
-    CLog::Log(LOGERROR, "CCurlFile::CReadState::Connect, didn't get any data from stream.");
-    return -1;
+    // Check response code
+    long response;
+    if (CURLE_OK == g_curlInterface.easy_getinfo(m_easyHandle, CURLINFO_RESPONSE_CODE, &response))
+      return response;
```

Comment:
This is actually not that nice. I'm not sure I understand the reason this was done, but a side-effect is that if you open an URL, get redirected, and an error occurs after the redirection (e.g. timeout), `Connect()` will pretend that everything is fine (since the response code is set). This causes the code that is downloading something to proceed as if it succeeded but it won't get any data.